portal-api/src/v1/gef/controllers/application-submit.js

```diff
@@ -28,7 +28,12 @@ const generateUkefDealId = async (application) => {
   };
 };
 
-const generateUkefFacilityId = (facility) => (facility.ukefFacilityId !== null ? facility.ukefFacilityId : generateId('facility'));
+const generateUkefFacilityId = async (facility) => {
+  const ukefFacilityId = facility.ukefFacilityId !== null ? facility.ukefFacilityId : await generateId('facility');
```

Comment:
Could this not be written as `facility.ukeffacilityId || await generateId('facility')` or is there a chance of a legitimate falsey value?

Reply:
Thanks @davehlewis - Yes, it probably could - when testing like that I got a failure, so added the explicit check.  I'll have a look at refactoring on the next PR.  This branch is a work in progress for the various steps on DTFS-4412

Reply:
No problem, can sometimes be a bit odd. If we could amend our eslint I would say nullish coalescing would work move cleanly for it.

Reply:
Updated.